index.js

```diff
@@ -720,30 +624,85 @@ TChannelConnection.prototype.send = function send(options, frame, callback) {
     //  throw new Error('duplicate frame id in flight');
     // }
 
-    frame.header.id = id;
-    frame.header.seq = 0;
+    var id = self.handler.sendRequestFrame({
+        arg1: arg1,
+        arg2: arg2,
+        arg3: arg3
+    }, function requestSent(err) {
+        if (err) {
+            self.logger.warn('failed to send request', {
+                id: id,
+                error: err
+            });
+            // TODO: retry?
```

Comment:
does tchannel do any network failure retries yet ? Does that belong in tchannel or tchannel_pool ?